### DISCUSSION TITLE
how to properly shutdown the server from shell script?

### DISCUSSION THREAD

**Discussion by cjbd** [gitpod-io/openvscode-server#503] [Q&A]
Hello, I'm creating a shell script to upgrading existing `openvscode-server` to latest version, I'm pretty closed to complete.
Right now, I'm able to get latest version from github, and running server version, compare version, download/extract/start the server

the problem i have is to stop existing server, i can use `kill -9 pid` but that only stops the shell process for `bin/openvscode-server` script, the spwned `node` processes, and `extension` processes are still there

so my question is how to properly shutdown the server, and all the spawned processed, using shell script

thank you!

**Comment by jeanp413** [comment]
I think just killing the `out/server-main.js` process instead should work

**Comment by cjbd** [comment]
@jeanp413 , thanks for the reply, let me try

**Reply by cjbd** [reply]
@jeanp413, that kills all node processes, there're still shell scripts for ternimals integration, i need to handle those too
